frontend/src/components/searchSelector/SearchSelector.tsx

```diff
@@ -0,0 +1,147 @@
+  searchHelpText?: string;
+  searchPlaceholder?: string;
+  searchValue: string;
+  toggleText: string;
+  toggleVariant?: React.ComponentProps<typeof MenuToggle>['variant'];
+};
+
+const SearchSelector: React.FC<SearchSelectorProps> = ({
+  children,
+  dataTestId,
+  isLoading,
+  isDisabled,
+  isFullWidth,
+  minWidth,
+  onSearchChange,
+  onSearchClear,
+  searchFocusOnOpen,
+  searchHelpText,
+  searchPlaceholder,
+  searchValue,
+  toggleText,
+  toggleVariant,
+}) => {
+  const [isOpen, setIsOpen] = React.useState(false);
+  const toggleRef = React.useRef(null);
+  const menuRef = React.useRef(null);
+  const searchRef = React.useRef<HTMLInputElement | null>(null);
+
+  return (
+    <MenuContainer
+      isOpen={isOpen}
+      menuRef={menuRef}
+      popperProps={{ minWidth, maxWidth: 'trigger' }}
+      onOpenChange={(open) => {
+        setIsOpen(open);
+        onSearchClear();
+      }}
+      toggleRef={toggleRef}
+      toggle={
+        <MenuToggle
+          id={dataTestId}
+          icon={
+            isLoading && (
+              <Icon>
+                <Spinner size="sm" aria-label="Loading" />
+              </Icon>
+            )
+          }
+          ref={toggleRef}
+          onClick={() => setIsOpen(!isOpen)}
+          isExpanded={isOpen}
+          isDisabled={isDisabled}
+          isFullWidth={isFullWidth}
+          data-testid={`${dataTestId}-toggle`}
+          variant={toggleVariant}
+        >
+          <TruncateNoMinWidth content={toggleText} />
+        </MenuToggle>
+      }
+      menu={
+        <Menu
+          data-testid={`${dataTestId}-menu`}
+          ref={menuRef}
+          isScrollable
+          onSelect={() => setIsOpen(false)}
+          onActionClick={() => setIsOpen(false)}
+        >
+          <MenuSearch>
+            <MenuSearchInput>
+              <SearchInput
+                ref={searchRef}
+                data-testid={`${dataTestId}-search`}
+                autoFocus={searchFocusOnOpen}
+                aria-label="Filter content"
+                onChange={(e, value) => onSearchChange(value)}
+                onClear={(e) => {
+                  e.stopPropagation();
+                  onSearchClear();
+                }}
+                placeholder={searchPlaceholder}
+                value={searchValue}
+              />
+            </MenuSearchInput>
+            {searchHelpText && (
+              <HelperText>
+                <HelperTextItem
+                  variant="indeterminate"
+                  data-testid={`${dataTestId}-searchHelpText`}
+                >
+                  {searchHelpText}
+                </HelperTextItem>
+              </HelperText>
+            )}
+          </MenuSearch>
+          <Divider />
+          <MenuContent>
+            <MenuList data-testid={`${dataTestId}-menuList`}>
+              {typeof children === 'function'
+                ? children({ menuClose: () => setIsOpen(false) })
+                : children}
```

Comment:
I'm not a fan of this -- but I feel there will always be rare cases that use "weirdly rendered" children -- in the case of Pipelines they render a table.

Now the alternative was to make "everyone" support he open state, but it seemed unnecessary messy API. So I made it possible to get it as a renderProp. 

Feedback welcome on this model, I could definitely adjust the whole component to a controlled vs uncontrolled nature too.

Reply:
for consistency maybe its better to convert the pipelines dropdowns to all follow the traditional menu items instead of having the table. All the table adds is sorting and being able to see and sort by creation date. Which im not sure is too helpfull? 

Then if we change that this can be moved to a fully controlled component. What do you think about creating a tech debt ticket to change that in pipelines and update this component to be controlled?

Reply:
Well, technically this model is not that foreign in React and it does still simplify things for the Pipeline group. I think Yan had intentional designs here for the awkwardness of the names of each version -- and Experiments & Pipelines sorta just followed suit. I'd hazard a guess that's all still true and removing this is less desirable. @Gkrumbach07 feel free to chase that with UX -- obviously a refactor in this PR is out of the question. If you want me to log something, I can, but it'll more likely be a UX ticket rather than a ENG ticket.

Reply:
all good.i can bring it up with UX at some point next week